code/datums/mind.dm

```diff
@@ -1463,6 +1466,15 @@
 		var/obj/effect/proc_holder/spell/S = X
 		S.action.Grant(new_character)
 
+/datum/mind/proc/disrupt_spells(delay, list/exceptions = New())
+	for(var/X in spell_list)
+		var/obj/effect/proc_holder/spell/S = X
+		for(var/type in exceptions)
+			if(istype(S, type))
+				continue
+		S.charge_counter = delay
+		addtimer(S, "start_recharge", 0)

```

Comment:
You'll want to use spawns here; our timer isn't precise enough to handle this, nor will an argument of 0 have the same impact as `spawn(0)` does on TG.